mlir/lib/Dialect/Vector/Transforms/VectorEmulateNarrowType.cpp

```diff
@@ -306,6 +307,73 @@ emulatedVectorLoad(OpBuilder &rewriter, Location loc, Value base,
       newLoad);
 }
 
+/// Atomically store a subbyte-sized value to memory, with a mask.
+static void atomicStore(OpBuilder &rewriter, Location loc,
+                        TypedValue<MemRefType> emulatedMemref,
+                        Value emulatedIndex, TypedValue<VectorType> value,
+                        Value mask, int64_t scale) {
```

Comment:
Couple nits:

- I'd rename it to builder instead of rewriter.
- `emulated*` seems to be a new terminology? Should we just use `linearized*` to avoid confusion?
- Should we use `numSrcElemsPerDest` instead of `scale`?

Reply:
Thanks for the suggestions! I think they improves readability.